packages/makeup-navigation-emitter/src/index.js

```diff
@@ -1,77 +1,186 @@
+
+    return items.indexOf(navigableItems[findIndexByAttribute(navigableItems, 'aria-selected', 'true')]);
+}
+
+function findIgnoredByDelegateItems(el, options) {
+    return (options.ignoreByDelegateSelector !== null) ?
+        [...el.querySelectorAll(options.ignoreByDelegateSelector)] : [];
+}
+
+function findPreviousNavigableIndex(items, index, wrap) {
+    let previousNavigableIndex = -1;
+
+    if (index === null) {
+        // no-op
+    } else if (atStart(items, index)) {
+        if (wrap === true) {
+            previousNavigableIndex = findLastNavigableIndex(items);
+        }
+    } else {
+        const previousSiblingItems = items.slice(0, index);
+        const allNavigableItems = findNavigableItems(items);
+        const previousNavigableSiblingItems = previousSiblingItems.reverse().filter(
+            el => allNavigableItems.includes(el)
+        );
+
+        if (previousNavigableSiblingItems[0]) {
+            previousNavigableIndex = items.indexOf(previousNavigableSiblingItems[0]);
         }
     }
+
+    return previousNavigableIndex;
 }
 
-function onKeyNext(e) {
-    if (isButton(e.detail.target) === false || this.options.ignoreButtons === false) {
-        if (!this.atEnd()) {
-            this.index++;
-        } else if (this.options.wrap) {
-            this.index = 0;
+function findNextNavigableIndex(items, index, wrap) {
+    let nextNavigableIndex = -1;
+
+    if (index === null) {
+        nextNavigableIndex = findFirstNavigableIndex(items);
+    } else if (atEnd(items, index)) {
+        if (wrap === true) {
+            nextNavigableIndex = findFirstNavigableIndex(items);
+        }
+    } else {
+        const nextSiblingItems = items.slice(index + 1);
+        const allNavigableItems = findNavigableItems(items);
+        const nextNavigableSiblingItems = nextSiblingItems.filter(
+            el => allNavigableItems.includes(el)
+        );
+
+        if (nextNavigableSiblingItems[0]) {
+            nextNavigableIndex = items.indexOf(nextNavigableSiblingItems[0]);
         }
     }
+
+    return nextNavigableIndex;
+}
+
+// returning -1 means not found
+function findIndexPositionByType(type, items, currentIndex) {
+    let index = -1;
+
+    switch (type) {
+        case 'none':
+            index = null;
+            break;
+        case 'current':
+            index = currentIndex;
+            break;
+        case 'interactive':
+            index = findFirstNavigableIndex(items);
+            break;
+        case 'ariaChecked':
+            index = findFirstNavigableAriaCheckedIndex(items);
+            break;
+        case 'ariaSelected':
+            index = findFirstNavigableAriaSelectedIndex(items);
+            break;
+        case 'ariaSelectedOrInteractive':
+            index = findFirstNavigableAriaSelectedIndex(items);
+            index = (index === -1) ? findFirstNavigableIndex(items) : index;
+            break;
+        default:
+            index = type;
```

Comment:
I see what you are doing here, but I think this is potentially cryptic behavior and could be the source for strange bugs since `type` is usually a string. It probably makes more sense to either throw an error here or to skip the default case.

Reply:
Agreed. Type is a poor choice of name. A new name and a type check will help. Typescript annotations are also going to help - I was thinking ahead to that. I do need to keep it as default case though as we need to support number being passed for backwards compatibility.